File: 020_retirement_py/retirement.py

def compute(start_age, initial_savings, info):
    curr_age = start_age
    curr_savings = initial_savings
    for i in range(info["months"]):
        curr_age_year = curr_age // 12
        curr_age_month = curr_age % 12
        print("Age  {} month{:3} you have ${:,.2f}".format(curr_age_year,curr_age_month,curr_savings))
        curr_savings =  curr_savings * (1+info["rate_of_return"]) + info["contribution"]
        curr_age += 1
    return curr_savings


def retirement(start_age, initial_savings, working_info,retired_info):
    """
    Prints the current status of an individual's retirement account.
    The dictionaries both have these fields: 
        "months","contribution","rate_of_return"

    Args:
       start_age (int): At what age (in months) does the individual start
       initial_savings (float): initial savings in dollars
       working_info (dict): information about working
       retired_into (dict): information about retirement

    Returns:
    None
    """

    #working
    curr_savings = compute(start_age,initial_savings,working_info)
    #retiring
    retire_start_age = start_age + working_info["months"]
    curr_savings = compute(retire_start_age,curr_savings,retired_info)
    return;
    



working_info = {"months":489,"contribution":1000,"rate_of_return":0.045/12}
retired_info = {"months":384,"contribution":-4000,"rate_of_return":0.01/12}
retirement(327,21345,working_info,retired_info)

File: 054_retirement/retirement.c
#include<stdlib.h>
#include<stdio.h>

struct _retire_info{
  int months;
  double contribution;
  double rate_of_return;
};
typedef struct _retire_info retire_info;

double compute_balance(double balance,retire_info retire){
  balance += balance * retire.rate_of_return + retire.contribution;
  return balance;
}


void retirement (int startAge,
		 double initial,
		 retire_info working,
		 retire_info retired
		 ){
  double balance = initial;
  for(int i = 0; i < working.months; i ++){
    printf("Age %3d month %2d you have $%.2f\n",((startAge+i)/12),((startAge+i)%12),balance);
    balance = compute_balance(balance, working);
  }
  startAge += working.months;
  for(int j = 0; j < retired.months; j ++){
    printf("Age %3d month %2d you have $%.2f\n",((startAge+j)/12),((startAge+j)%12),balance);
    balance = compute_balance(balance, retired);
  }  
}


int main(void){;
  retire_info working_info;
  working_info.months = 489;
  working_info.contribution  = 1000;
  working_info.rate_of_return = 0.045/12;
  retire_info retired_info;
  retired_info.months = 384;
  retired_info.contribution = -4000;
  retired_info.rate_of_return = 0.01/12;
  retirement(327,21345,working_info,retired_info);
  return 1;
}




